### ISSUE TITLE
Default argument of `converter` doesn't work in field specifiers of `dataclass_transform`

### ISSUE THREAD

**Issue by PragmaTwice** [microsoft/pyright#11291]
**Describe the bug**

In field specifiers of `dataclass_transform`, we can provide a `converter` parameter to control the parameter type in `__init__`. In the following example, we provide a `converter=foo` where `def foo(x: int) -> str`, so in `__init__` `x` should be typed `int` instead of `str`.

```python
from typing import dataclass_transform

def foo(x: int) -> str: ...

def field(*, converter):
    pass

@dataclass_transform(field_specifiers=(field,))
class A:
    @classmethod
    def __init_subclass__(cls):
        pass

class B(A):
    x: str = field(converter=foo)

# __init__ signature: B(x: int)
```

But if we move the `converter=foo` to a default argument of `converter` in `def field(..)`, the effect of `converter` will disappear. This is not consistent with other parameters of field specifiers, as they can all be a default argument instead of being an explicit argument.

```python
from typing import dataclass_transform

def foo(x: int) -> str: ...

def field(*, converter=foo):
    pass

@dataclass_transform(field_specifiers=(field,))
class A:
    @classmethod
    def __init_subclass__(cls):
        pass

class B(A):
    x: str = field()

# wrong __init__ signature: B(x: str). should be B(x: int) instead
```

**VS Code extension or command-line**

ms-python.vscode-pylance 2026.1.1

**Comment by erictraut** [comment]
Pyright is working as intended here. The `dataclass_transform` mechanism (described in [the Python typing spec](https://typing.python.org/en/latest/spec/dataclasses.html)) does not support field descriptors with default arguments in this manner. If you want your code to type check with pyright or other standards-compliant type checkers, you should use `converter=foo` directly in the field constructor call.

**Comment by PragmaTwice** [comment]
> The dataclass_transform mechanism (described in [the Python typing spec](https://typing.python.org/en/latest/spec/dataclasses.html)) does not support field descriptors with default arguments in this manner.

Really? Then how you explain this code example inside the spec?

```python
# Library code (within type stub or inline)
# In this library, passing a resolver means that init must be False,
# and the overload with Literal[False] enforces that.
@overload
def model_field(
        *,
        default: Optional[Any] = ...,
        resolver: Callable[[], Any],
        init: Literal[False] = False,
    ) -> Any: ...

@overload
def model_field(
        *,
        default: Optional[Any] = ...,
        resolver: None = None,
        init: bool = True,
    ) -> Any: ...

@typing.dataclass_transform(
    kw_only_default=True,
    field_specifiers=(model_field, ))
def create_model(
    *,
    init: bool = True,
) -> Callable[[Type[_T]], Type[_T]]: ...

# Code that imports this library:
@create_model(init=False)
class CustomerModel:
    id: int = model_field(resolver=lambda : 0)
    name: str
```

Note that `init` has default parameters, and it works well in pyright. i.e. pyright respect these default parameters. I think that the behavior of `init` and `converter` should be consistent.

Refer to https://typing.python.org/en/latest/spec/dataclasses.html#field-specifier-parameters.
